python/packages/ai/teams/ai/planner/openai_planner.py

```diff
@@ -144,25 +153,71 @@ async def _complete_prompt(
 
         result: str
 
-        # TODO: implement chat completion
-        result = await self._complete_text(prompt_template)
+        if is_chat_completion:
+            user_message = turn_state.temp.value.input
+            result = await self._complete_chat(
+                prompt_template, turn_state, user_message, history_options
+            )
+        else:
+            result = await self._complete_text(prompt_template)
 
         duration = datetime.now() - start_time
         # TODO: investigate how to get prompt/completion tokens
         self._log_request(f"\n{log_prefix} SUCCEEDED: duration={duration} response={result}")
 
         return result
 
+    async def _complete_chat(
+        self,
+        prompt_template: PromptTemplate,
+        state: TurnState,
+        user_message: str,
+        options: AIHistoryOptions,
+    ):
+        prompt_template_config = generate_sk_prompt_template_config(prompt_template)
+        request_settings = ChatRequestSettings.from_completion_config(
+            prompt_template_config.completion
+        )
+
+        chat_completion_client = self._sk.get_ai_service(ChatCompletionClientBase)(self._sk)
```

Comment:
this doesn't work, throws an error when run in a bot

Reply:
The current code in this PR (which removes the extra logger parameter) works.

The type checking warning you mentioned in my previous PR should be ignored (and mypy in our CI does not complain this actually), because of a Semantic Kernel defect. If you look into the implementation, it actually doesn't implement the exact interface defined in `ChatCompletionClientBase` and caused the type checking error.

I have raised an issue to their repo and once they fixed the issue, the warning should be gone.